src/http_protocol.cpp

```diff
@@ -460,10 +461,11 @@ http_request_t::from_stream(const char* start, size_t length, size_t max_size) {
           // while the char is in the set, basically implement TRIM.. lets instead hope to ditch our
           // custom parser
           size_t field_end, value_begin;
-          if ((field_end = partial_buffer.find(':')) == std::string::npos ||
-              (value_begin = partial_buffer.find_first_not_of(' ', field_end + 1)) ==
-                  std::string::npos)
+          if ((field_end = partial_buffer.find(':')) == std::string::npos)
             throw RESPONSE_400;
+          if ((value_begin = partial_buffer.find_first_not_of(' ', field_end + 1)) ==
+              std::string::npos)
+            value_begin = partial_buffer.size();
```

Comment:
but now if you dont send a value or if your value is all whitespace we detect that and set the value to empty string